third_party/ti_simplelink_sdk/ti_simplelink_executable.gni

```diff
@@ -37,6 +37,17 @@ assert(matter_software_ver_str != "",
 template("ti_simplelink_executable") {
   simplelink_target_name = target_name
 
+  # Run prebuild SDK driver generation
+  if (ti_simplelink_device_family == "cc13x2x7_26x2x7" ||
```

Comment:
Will this be run during the gen phase?

I would expect instead that we only run scripts during compile phase, especially for things athat are basically a `make`

Reply:
@andy31415 That was the attempted solution, but we ran into an issue where Ninja was unable to find the path to the (Not yet generated) library; "ninja: error: '/path/to/samplelib.a', needed by 'project', missing and no known rule to make it
".
Normally for CMAKE this would be resolved by specifying the library within a BYPRODUCTS section of the build. Unfortunately, I have not found an equivalent solution when using GN/NINJA.

How would you recommend building external dependencies when the project requires that all exist in the project during the compile phase?

Reply:
you would need to make an action and declare its outputs at least as samplelib.a and then make the executable or whatever depend on samplelib depend on the action.

Reply:
Hi @andy31415, I've reviewed the code base extensively and have attempted the following:

		
		1. Create explicit dependency on the library to be generated by run_sdk_drivers_gen.py to be run at compile time.
			i. GN/Ninja does not allow libraries to be listed as outputs for actions, only c/header files are allowed.
			ii. I've attempted to work around this issue by generating an intermediary 'dummy.c' file that is listed as a dependency for all targets, but this has also not yielded successful results.
		2. Manually set all source files in library within application
			i. Not feasible as these libraries contain specific rules that would need to be maintained/duplicated separately within the Matter repository
			ii. Not optimal as this would require users to rebuild the library per example, compared to once per session
I'm at an impasse as I'm unable to conjure up a valid solution based on your feedback. Is there another route/secret ninja command I may be missing to resolve this issue? If not, are there any other barriers from merging the PR apart from the generation phase generally not running these scripts?